browser/src/app/TilesMiddleware.ts

```diff
@@ -889,11 +889,12 @@ class TileManager {
 
 		// Don't paint the tile, only dirty the sectionsContainer if it is in the visible area.
 		// _emitSlurpedTileEvents() will repaint canvas (if it is dirty).
-		if (
-			app.isRectangleVisibleInTheDisplayedArea(
-				this.pixelCoordsToTwipTileBounds(coords),
-			)
-		)
+		const tileVisible = app.isRectangleVisibleInTheDisplayedArea(
+			this.pixelCoordsToTwipTileBounds(coords),
+		);
+
+		// Also check the scale because incoming tile may not be in the same zoom level.
+		if (tileVisible && coords.scale === app.getScale())
```

Comment:
I'm not keen on the floating point comparison here (I realise it ought to work given the rest of the changes, but we're relying on external, undocumented properties never changing) - I'd rather we didn't change `app.getScale` (which seems prone to break other things/cause interesting to track errors), multiplied the scale out here and compared the integer values.

Reply:
That's right, somehow i picked the other way but this is better. Improved.